pkg/client/client.go

```diff
@@ -59,3 +68,16 @@ func NewClient(kubeConfig, kubeContext string) (*Client, error) {
 		HarvesterNetworkClient:    harvNetworkClient,
 	}, nil
 }
+
+func restConfigFromFile(kubeConfig, kubeContext string) (*rest.Config, error) {
```

Comment:
Will it make sense to move the `homedir.Expand(d.Get(constants.FieldProviderKubeConfig).(string))` at https://github.com/harvester/terraform-provider-harvester/blob/35e2902d29e6590b80bb8e05b9c1af41f5aa1ca9/internal/provider/provider.go#L71-L74 into this new function?

Reply:
Yeah. That would make sense.

`homedir.Expand` really on expands the `~` to whatever the invoking user's home directory path is. It's only relevant if you're dealing with a path, not if you're dealing with base64 encoded strings, since `~` can't occur in base64 data.

I wonder if there isn't a better more canonical way to do this only with stuff from the Golang standard library. We're taking on a dependency only for this and the dependency isn't maintained anymore. Not a big deal, since it's not code that's likely to blow up in our face anytime soon, but still not good.
But we do get some portability for MacOS and Windows from this library, so ripping it out isn't necessarily trivial - depending on what the state of portability for Golang is nowadays.

Reply:
Yeah, I am not fond of that dependency. I'm betting it came from some terraform example code. Personally, I will just make it an absolute path, and enforce it with `filepath.IsAbs()`.